GitCommands/Git/GitBlame.cs

```diff
@@ -1,116 +1,81 @@
 ﻿using System;
 using System.Collections.Generic;
-using System.Drawing;
-using System.Linq;
+using System.Globalization;
 using System.Text;
+using JetBrains.Annotations;
 
 namespace GitCommands
 {
-    public class GitBlame
+    public sealed class GitBlame
     {
-        public GitBlame()
-        {
-            Headers = new List<GitBlameHeader>();
-            Lines = new List<GitBlameLine>();
-        }
+        public IReadOnlyList<GitBlameLine> Lines { get; }
 
-        public IList<GitBlameHeader> Headers { get; }
-        public IList<GitBlameLine> Lines { get; }
-
-        public GitBlameHeader FindHeaderForCommitGuid(string commitGuid)
+        public GitBlame(IReadOnlyList<GitBlameLine> lines)
         {
-            return Headers.First(h => h.CommitGuid == commitGuid);
```

Comment:
This is the O(N) function I referred to. Now each line has a direct reference to the commit object, and the commit objects are shared between lines as appropriate.